src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -19,17 +19,27 @@
 
 #include <chrono>
 
+using namespace std::chrono_literals;
+
 namespace pandaproxy {
 
+static constexpr auto gc_timer_period = 10s;
+
 kafka_client_cache::kafka_client_cache(
-  YAML::Node const& cfg,
-  size_t max_size,
-  std::chrono::milliseconds keep_alive,
-  ss::timer<ss::lowres_clock>& clean_and_evict_timer)
+  YAML::Node const& cfg, size_t max_size, std::chrono::milliseconds keep_alive)
   : _config{cfg}
   , _cache_max_size{max_size}
-  , _keep_alive{keep_alive}
-  , _clean_and_evict_timer{clean_and_evict_timer} {}
+  , _keep_alive{keep_alive} {
+    _gc_timer.set_callback([this] {
+        ssx::spawn_with_gate(_gc_gate, [this] {
+            return clean_stale_clients().finally([this] {
+                if (!_gc_gate.is_closed()) {
+                    _gc_timer.rearm(ss::lowres_clock::now() + gc_timer_period);
```

Comment:
I ran into a failure where the timer was `arm`'d twice causing an assert failure at the seastar level. Using `rearm` here protects against that because it cancels the timer first. I can put this in a comment.